honeybee_vtk/data.py

```diff
@@ -0,0 +1,161 @@
+"""Data json schema and validation."""
+
+import json
+from json.decoder import JSONDecodeError
+from pathlib import Path
+from typing import Dict, List
+from pydantic import BaseModel, validator, Field
+from pydantic.errors import JsonError
+from .model import Model
+from .types import AcceptedValues
+
+
+class Data(BaseModel):
+    """COnfig for data."""
+
+    name: str = Field(
+        description='Name to be give to data. Example, "Daylight-Factor".'
+    )
+
+    object_name: str = Field(
+        description='The name of the model object on which you would like to map this'
+        ' data.'
+    )
+
+    file_paths: List[str] = Field(
+        description='List of paths to the file or files that you are trying to use as'
+        ' data.'
+    )
+
+    @validator('object_name')
+    def validate_object(cls, v: str) -> str:
+        if v in AcceptedValues.names.value:
+            return v
+        else:
+            raise ValueError(
+                f'Object name should be from these {AcceptedValues.names.value}.'
+                f' Instead got {v}.'
+            )
+
+    @validator('file_paths')
+    def validate_paths(cls, v: List[str]) -> List[str]:
+        if all([Path(path).is_file() for path in v]):
+            return v
+        else:
+            raise ValueError(
+                'File paths are not valid.'
+            )
+
+    def cross_check_data(self, model: Model) -> bool:
+        """Cross check data with model.
+
+        For grids, it will be checked if the number of data files and the names of the
+        data files match with the grid identifiers. For other than grid objects, it will
+        be checked that only one data file is provided and the length of the data file
+        matches the length of Polydata in the model.
+
+        Args:
+            model: A honeybee-vtk model object.
+
+        Returns:
+            A boolean value.
+        """
+
+        # if object name is "grid" check that the name of files match the grid names
+        if self.object_name == AcceptedValues.names.value[-1]:
+
+            assert len(model.sensor_grids.data) > 0, 'Sensor grids are not loaded on'\
+                ' this model. Reload them using grid options.'
+
+            grid_names = [grid.identifier for grid in model.sensor_grids.data]
+            file_names = [Path(path).name.split('.')[0] for path in self.file_paths]
+            if len(grid_names) != len(file_names) or grid_names != file_names:
+                raise ValueError(
+                    'The number of files and the file names must match the grid'
+                    ' identifiers in HBJSON.'
+                )
```

Comment:
These are two different checks and the error message should tell which one has failed. I would do two checks.

Also what if I want to load the results only for a single grid in the model? That's considered invalid?

Reply:
> Also what if I want to load the results only for a single grid in the model? That's considered invalid?

I don't think it should be invalid. I will have to make some changes to how designed add_data_fields method in ModelDataSet. I will do it.

Reply:
If it is two much work for now then ignore this comment and document it as a limitation.

Reply:
I spent two hours on this and couldn't do it. I created an issue to do this later. https://github.com/ladybug-tools/honeybee-vtk/issues/160

Reply:
Good strategy!